pkg/services/serviceaccounts/database/store.go

```diff
@@ -467,9 +466,6 @@ func (s *ServiceAccountsStoreImpl) CreateServiceAccountFromApikey(ctx context.Co
 		}
 
 		if err := s.assignApiKeyToServiceAccount(sess, key.Id, newSA.ID); err != nil {
-			if err := s.userService.Delete(ctx, &user.DeleteUserCommand{UserID: newSA.ID}); err != nil {
```

Comment:
since we are in a transaction the idea is that the created user in line [464](https://github.com/grafana/grafana/pull/58961/files#diff-8a7c4a45e4d47c36f84d571baa67813c222fc5cd8d02ed9b34bf4fd3e711eb90L464) will be deleted on the rollback, since we have transaction rolling from 463.